main.go

```diff
@@ -90,29 +90,49 @@ func processCommand() int {
 	if !controlConnection && options.OpenSSH.ControlCommand != "exit" {
 		updateRequestExtensions(&vaultClient.Options.Extensions, &sshClient.Extensions)
 
-		signedKey, err := vaultClient.GetSignedKey(sshClient.User)
-		if err != nil {
-			log.Fatal("failed to get signed key: ", err)
+		log.Debugf("running in %q mode\n", vaultClient.Options.Mode)
+		switch vaultClient.Options.Mode {
+		case "issue":
+			agent, err := agent.NewInternalAgent()
+			if err != nil {
+				log.Fatal("failed to start internal agent: ", err)
+			}
+			defer agent.Stop()
+
+			privateKey, signedKey, err := vaultClient.GenerateSignedKeypair(sshClient.User)
+			if err != nil {
+				log.Fatal("failed to generate signed keypair: ", err)
+			}
+
+			if err := agent.AddSignedKeyPair(privateKey, signedKey); err != nil {
+				log.Fatal("failed to add keypair to internal agent: ", err)
+			}
+
+			// override default ssh-agent socket
+			os.Setenv("SSH_AUTH_SOCK", agent.SocketFile())
```

Comment:
Would it be a good idea to store the existing `SSH_AUTH_PROXY` value so that, upon completion of function, the original user-defined value is restored?

Reply:
We only override within this process, not in the parent environment, so this is unnecessary.